go/db_factory.go

```diff
@@ -42,9 +46,53 @@ func (f *TrinoDBFactory) CreateDB(ctx context.Context, driverName string, opts m
 		return nil, err
 	}
 
+	// Register custom HTTP client with 10-hour timeout to prevent infrastructure timeouts
+	dsn, err = f.registerCustomClientForTimeout(dsn)
+	if err != nil {
+		return nil, err
+	}
+
 	return sql.Open(driverName, dsn)
 }
 
+// registerCustomClientForTimeout creates and registers a custom HTTP client with a 10-hour timeout.
+// This prevents infrastructure/network timeouts during long-running queries
+func (f *TrinoDBFactory) registerCustomClientForTimeout(dsn string) (string, error) {
+	cfg, err := trino.ParseDSN(dsn)
+	if err != nil {
+		return "", fmt.Errorf("failed to parse DSN: %v", err)
+	}
+
+	const httpClientTimeout = 10 * time.Hour
+
+	customClient := &http.Client{
+		Timeout: httpClientTimeout, // Overall request timeout
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+				DualStack: true,
+			}).DialContext,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ResponseHeaderTimeout: httpClientTimeout, // Timeout for response headers
+			ExpectContinueTimeout: 1 * time.Second,
+		},
```

Comment:
Is it possible to avoid hardcoding the rest of these constants and let them take whatever the default value is?